app/controllers/redirection_controller.rb

```diff
@@ -25,4 +25,27 @@ def publications
       format.atom { redirect_to(finder_path('search/all', params: parameters, format: :atom)) }
     end
   end
+
+  def published_statistics
+    parameters = { keywords: params['keywords'],
+                  level_one_taxon: params['taxons'].try(:first),
+                  organisations: params['departments'],
+                  public_timestamp: { from: params['from_date'], to: params['to_date'] }.compact.presence }
+    respond_to do |format|
+      format.html { redirect_to(finder_path('search/statistics', params: parameters)) }
+      format.atom { redirect_to(finder_path('search/statistics', params: parameters, format: :atom)) }
+    end
+  end
+
+  def upcoming_statistics
+    parameters = { keywords: params['keywords'],
+                  level_one_taxon: params['topics'].try(:first),
+                  organisations: params['organisations'],
+                  content_store_document_type: :statistics_upcoming,
+                  public_timestamp: { from: params['from_date'], to: params['to_date'] }.compact.presence }
```

Comment:
As this logic is nearly identical to `published_statistics` I'd suggest moving the common logic to a separate method eg 
```
def convert_statistics_parameters
{ keywords: params['keywords'],
                  level_one_taxon: params['taxons'].try(:first),
                  organisations: params['departments'],
                  public_timestamp: { from: params['from_date'], to: params['to_date']
}.compact.presence }
end
```
and then doing something like this in upcoming statistics
```
parameters = convert_statistics_parameters.merge({ content_store_document_type: :statistics_upcoming})
```

which keeps things a little neater

Reply:
Alright, I've kind of ended up implementing the opposite of this, where every other method except upcoming_statistics uses a common_parameters method to handle stuff they're all using. This is because the upcoming stats page uses it's own special parameter names for some reason, calling things 'topics' rather than 'taxons' and 'departments' rather than 'organisations'.

But the main efficiency is there, and it'll be much easier to add more methods now it doesn't have to get retyped every time. 

I'll find out from someone how to make it all one big commit.

Reply:
What you could do is also move this stuff to the params method. Where you have multiple keys being used, like organisations / departments both being the same thing, you could do this:

```
organisations: params['organisations'] || params['departments'],
```